Personnummer/Personnummer.cs

```diff
@@ -1,227 +1,198 @@
-            }
-
-            return valid && Luhn($"{parsed.Decade}{parsed.Month}{parsed.Day}{parsed.Digits}") == 0;
-
-        }
-
-        /// <summary>
-        /// Format a Swedish social security number in long format to a string format.
-        /// </summary>
-        /// <param name="value">Value as long.</param>
-        /// <param name="withCentury">Include century instead of +/-.</param>
-        /// <returns>Swedish social security number as a string.</returns>
-        /// <exception cref="ValidationException">On invalid social security or coordination number.</exception>
-        public static string Format(long value, bool withCentury = false)
-        {
-            ParsedPersonnummer parsed = Parse(value.ToString());
-            if (!parsed.Parsed ||  Luhn($"{parsed.Decade}{parsed.Month}{parsed.Day}{parsed.Digits}") != 0)
-            {
-                throw new ValidationException($"{value} is not a valid Swedish social security or coordination number.");
-            }
-
-            if (withCentury)
-            {
-                return $"{parsed.Century}{parsed.Decade}{parsed.Month}{parsed.Day}{parsed.Digits}";
-            }
-
-            char sign = ((DateTime.Now.Year - int.Parse(parsed.Century + parsed.Decade)) >= 100) ? '+' : '-';
-            return $"{parsed.Decade}{parsed.Month}{parsed.Day}{sign}{parsed.Digits}";
-        }
-
-        /// <summary>
-        /// Format a Swedish social security number in string format to a long format.
-        /// </summary>
-        /// <param name="value">Value as string.</param>
-        /// <param name="withCentury">Include century in output value.</param>
-        /// <returns>Swedish social security number as a long.</returns>
-        /// <exception cref="ValidationException">On invalid social security or coordination number.</exception>
-        public static long Format(string value, bool withCentury = false)
-        {
-            ParsedPersonnummer parsed = Parse(value);
-            if (!parsed.Parsed || Luhn($"{parsed.Decade}{parsed.Month}{parsed.Day}{parsed.Digits}") != 0)
-            {
-                throw new ValidationException($"{value} is not a valid Swedish social security or coordination number.");
-            }
-
-            if (withCentury)
-            {
-                return long.Parse($"{parsed.Century}{parsed.Decade}{parsed.Month}{parsed.Day}{parsed.Digits}");
-            }
-
-            return long.Parse($"{parsed.Decade}{parsed.Month}{parsed.Day}{parsed.Digits}");
-        }
-    }
-}
+﻿using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Personnummer.Exceptions;
+
+namespace Personnummer
+{
+    public struct PersonnummerOptions
+    {
+        #region Fields and Properties
+
+        public bool AllowCoordinationNumber { get; set; }
+
+        #endregion
+    }
+
+    public class Personnummer
+    {
+        #region Fields and Properties
+
+        public string Age { get; }
+

```

Comment:
We changed from `Age` property to `GetAge` method, see https://github.com/personnummer/php/commit/290471d7ff440a6ced9b45b7aeab000147331df7

Reply:
Should we do that with C# too? Seeing "get" methods in c# is quite against the standard?

Reply:
I honestly can't remember why we changed it to a `GetAge`, could you remind me?

Reply:
We change it because of that we shouldn't parse the age before needing it and because of the int vs string discussion. If you think it's better to name the method just Age or something I think it's okay to do it.

Reply:
Well, I'm absolutely okay with it being computed in realtime instead of in the constructor, and I thought about that before too, hehe, but when it comes to the `Age` vs `GetAge` I'm feeling that C# (as an exception) should probably use a Age property (with get only and computed in realtime) while other languages who don't utilize properties in the same way, might be better off with a `get` method.  
I'm open for not using a property though, so depends on how you feel about it? :)

Reply:
Updated Age and Separator properties to compute in realtime. If we want it to be a `GetAge` method instead, I'll fix that too. Let me know how you feel :)

Reply:
I'm okay with not having `Get` in the method name, not sure about the compute since all other implementations have method. We should return `age` as a int and not a string. 

ping @rasmusbe on the compute, name and method things.

Reply:
Oh, I meant that it will "behave" as a method in the get :)
```
        public string Age => (DateTime.Now.Year - (new DateTime(int.Parse($"{FullYear}"), int.Parse(Month), realDay, 0, 0, 0)).Year).ToString();
        public string Separator => int.Parse(Age) >= 100 ? "+" : "-";
```

Reply:
Ah nice, just change `Age` to int then and we will be good to go!

Reply:
Super duper, preferred it as int also ;P

All done! :)